File: sdk-testing/src/sdk_testing/langchain_demo.py
import os
from pathlib import Path

from dotenv import load_dotenv
import neatlogs

load_dotenv(Path(__file__).resolve().parents[2] / ".env")

neatlogs.init(
    api_key=os.environ["NEATLOGS_API_KEY"],
    endpoint=os.environ.get("NEATLOGS_ENDPOINT", "http://localhost:4100"),
    workflow_name="langchain-demo",
    instrumentations=["langchain"],
)

# Import LangChain AFTER init() so it's patched.
from langchain_openai import ChatOpenAI

llm = ChatOpenAI(model="gpt-4o")

print(llm.invoke("In one sentence, what is LangChain?").content)

neatlogs.flush()
neatlogs.shutdown()

File: sdk-testing-ts/src/langchain_demo.ts
import { config } from "dotenv";
import { resolve } from "path";
import { init, langchainHandler, flush, shutdown } from "neatlogs";

config({ path: resolve(import.meta.dir, "../../sdk-testing/.env") });

async function main() {
  await init({
    apiKey: process.env.NEATLOGS_API_KEY,
    endpoint: process.env.NEATLOGS_ENDPOINT,
    workflowName: "langchain-demo",
  });

  const { ChatOpenAI } = await import("@langchain/openai");
  const llm = new ChatOpenAI({ model: "gpt-4o" });

  const handler = langchainHandler();
  const res = await llm.invoke("In one sentence, what is LangChain?", {
    callbacks: [handler],
  });
  console.log(res.content);

  await flush();
  await shutdown();
}

main().catch(console.error);
